core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/TerminationService.java

```diff
@@ -148,4 +154,11 @@ private void terminateMetaDataInstances(Stack stack) {
         instanceMetaDataService.saveAll(instanceMetaDatas);
     }
 
+    private void deleteMetaDataInstances(Stack stack) {
+        for (InstanceMetaData metaData : stack.getNotDeletedInstanceMetaDataSet()) {
+            LOGGER.debug("Deleting instance metadata entry {}", metaData);
+            instanceMetaDataService.delete(metaData);
```

Comment:
I remember now why delete is needed: because the new instances were brought up with exactly the same metadata causing DB key collision if we'd keep the original row and try to create a new one.

Reply:
I don't understand how. Only thing with unique index is the ID field, that shouldn't be a problem. Private id is also increased, so that should work.

Reply:
I might have been wrong in DB keys, but PrivateId-s need to be reassigned from 0 here to have the original instanceNames. 
Also these  finders (there are 6 of theses not restricted to not terminated instances) should be revised, if I'd keep these instancemetadata-s, I'd refrain testing all of those in this PR-s context:

```
    @EntityGraph(value = "InstanceMetaData.instanceGroup", type = EntityGraphType.LOAD)
    @Query("SELECT i FROM InstanceMetaData i WHERE i.instanceId= :instanceId AND i.instanceGroup.stack.id= :stackId")
    Optional<InstanceMetaData> findByStackIdAndInstanceId(@Param("stackId") Long stackId, @Param("instanceId") String instanceId);
```